docs/DafnyRef/Attributes.md

```diff
@@ -344,136 +343,217 @@ recursion was explicitly requested.
+
+To achieve higher verification performance, Dafny collects all the assertions of one method into one single conjecture (an _assertion batch_) that it sends to the verifier, which tries to prove it correct:
+
+* If the verifier says it is correct[^smt-encoding], it means that all the assertions hold.
+* If the verifier returns a counter-example, this counter-example is used to determine both the failing assertion and the failing path.
+  Dafny will then query again the verifier with the assumption that that particular assert is valid, so that it can retrieve other failing assertions happening afterwards.[^caveat-about-assertion-and-assumption]
+* If the verifier returns `unknown` or times out, or even preemptively for difficult assertions or to reduce the chance that the verifier will ‘be confused’ by the many assertions in a large batch, Dafny partitions the assertions up into smaller batches[^smaller-batches]. An extreme case is the use of the `/oneAssertAtATime` command-line option, which causes Dafny to make one batch for each assertion.
+
+[^smt-encoding]: The formula sent to the underlying SMT solver is the negation of the formula that the verifier wants to prove. Hence, if the SMT solver returns "unsat", it means that the SMT formula is always false, meaning the verifier's formula is always true. On the other side, if the SMT solver returns "sat", it means that the SMT formula can be made true with a special variable assignment, which means that the verifier's formula is false under that same variable assignment, meaning it's a counter-example for the verifier.
+
+[^caveat-about-assertion-and-assumption]: Caveat about assertion and assumption: One big difference between an "assertion transformed in an assumption" and the original "assertion" is that the original "assertion" can unroll functions twice, whereas the "assumed assertion" can unroll them only once. Hence, Dafny can still continue to analyze assertions after a failing assertion without automatically proving "false" (which would make all further assertions to prove automatically).
+
+[^smaller-batches]: To create a smaller batch, Dafny duplicates the list of assertions, and ensure every assertion is transformed into an assumption in exactly one list, so that the assertions of the two new lists form a partition of the original list. This results in "easier" formulas for the verifier because it has less to prove, but it takes more overhead because every verification instance have a common set of axioms and there is no knowledge sharing between instances because they run independently.
+
+This is where the following annotations `{:focus}` and `{:split_here}` come it.
+They enables to manually force the verifier to split assertions into two lists, so that both of them can verify usually independently faster than the original problem, although the total verification time, if not using parallel cores, can double at most.
+
```

Comment:
I'm not sure it's true that it can double at most. In some weird cases, I think it's possible for splitting verification into batches to slow things down dramatically. Usually it's not worse than doubling (and often it's much better), but I think it's possible for it to be.

Reply:
I replaced `can double at most.` with `might increase.`

Reply:
Sounds good!